src/sharing/ShareToPm.js

```diff
@@ -169,7 +153,7 @@ class ShareToPm extends React.Component<Props, State> {
 
     if (choosingRecipients) {
       return (
-        <Modal>
+        <Modal onRequestClose={this.handleModalClose}>
```

Comment:
sharing: Handle hardware back button when selecting PM recipients.
    
    Use modal's 'onRequestClose' to hide the user selection screen when
    the back button is pressed while choosing recipients in the share to
    PM tab.

Hmm, did this not work in the previous version? Certainly good to fix so that this works.

I'm less happy about relying completely on the back button here. It's acceptable as a temporary workaround because we only support this flow in the first place on Android -- but that won't work when in the future we offer sharing on iOS as well. And in the UX of this even on Android: everywhere else in the app (including other uses of this user-picker widget!), when it's possible and natural to go back, we have a "back" icon at the top left of the screen that offers that. It feels like a glitch for that to be missing here.